### PR TITLE
add sphinx conf to auto generate python api doc

### PR DESCRIPTION
**Pull request by Nicole00**
run docs/build.sh to auto generate the nebula-python api doc.
the doc looks like:
image

### CODE REVIEW

**Comment by wey-gu** [comment]
This is a great milestone.

We could leverage https://github.com/pradyunsg/furo for better theming and then add more .md/ jupyter dev docs/examples afterwards.

File: docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Configuration file for the Sphinx documentation builder.
```
Comment (wey-gu) [inline docs/source/conf.py:1]:
```
make fmt-check
```
linting failures

**Review by wey-gu — DISMISSED** [review]

**Review by wey-gu — DISMISSED** [review]

File: docs/build.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+pip install--user sphinx
```
Comment (wey-gu) [inline docs/build.sh:1]:
```diff
- pip install--user sphinx
+ pip install --user sphinx

Comment (wey-gu) [reply docs/build.sh:1]: We could leverage the state-of-art best theme now?

Install Furo in the documentation's build environment.

```bash
pip install install --user sphinx furo
```

Then update the html_theme in conf.py.

```python
html_theme = "furo"
```

ref: https://github.com/pradyunsg/furo

furo theme will look like https://devguide.python.org/

Comment (wey-gu) [reply docs/build.sh:1]: I can prepare for some guides later with this as the base! And afterwards, we could at some point publish this to readthedocs.io

**Review by wey-gu — DISMISSED** [review]
GREAT JOB!

**Comment by wey-gu** [comment]
Seems `resp.space_name() == ''` is expected in current master/nightly-build of NebulaGraph core?

```python
    def test_2_reconnect(self):
        try:
            session = self.pool.get_session('root', 'nebula')
            time.sleep(2)
    
            # wait for the session space info to be updated to meta service
            session.execute(
                'CREATE SPACE IF NOT EXISTS test_session(vid_type=FIXED_STRING(8)); USE test_session;'
            )
            time.sleep(10)
            for i in range(0, 5):
                if i == 3:
                    os.system('docker stop tests_graphd0_1')
                    os.system('docker stop tests_graphd1_1')
                    time.sleep(3)
                resp = session.execute('SHOW SESSIONS')
                assert resp.is_succeeded(), resp.error_msg()
                assert resp.space_name() == 'test_session' #<---------------------
                time.sleep(2)
            session.release()
            new_session = self.pool.get_session('root', 'nebula')
            new_session.execute('SHOW SPACES')
        except Exception as e:
>           assert False, e
E           AssertionError: AssertionError("assert '' == 'test_session' #-----------
E               - test_session") 
E           assert False

```

If that's the case, let's remove this assert from UT.

**Comment by Nicole00** [comment]
> Seems `resp.space_name() == ''` is expected in current master/nightly-build of NebulaGraph core?
> 
> ```python
>     def test_2_reconnect(self):
>         try:
>             session = self.pool.get_session('root', 'nebula')
>             time.sleep(2)
>     
>             # wait for the session space info to be updated to meta service
>             session.execute(
>                 'CREATE SPACE IF NOT EXISTS test_session(vid_type=FIXED_STRING(8)); USE test_session;'
>             )
>             time.sleep(10)
>             for i in range(0, 5):
>                 if i == 3:
>                     os.system('docker stop tests_graphd0_1')
>                     os.system('docker stop tests_graphd1_1')
>                     time.sleep(3)
>                 resp = session.execute('SHOW SESSIONS')
>                 assert resp.is_succeeded(), resp.error_msg()
>                 assert resp.space_name() == 'test_session' #<---------------------
>                 time.sleep(2)
>             session.release()
>             new_session = self.pool.get_session('root', 'nebula')
>             new_session.execute('SHOW SPACES')
>         except Exception as e:
> >           assert False, e
> E           AssertionError: AssertionError("assert '' == 'test_session' #-----------
> E               - test_session") 
> E           assert False
> ```
> 
> If that's the case, let's remove this assert from UT.



> Seems `resp.space_name() == ''` is expected in current master/nightly-build of NebulaGraph core?
> 
> ```python
>     def test_2_reconnect(self):
>         try:
>             session = self.pool.get_session('root', 'nebula')
>             time.sleep(2)
>     
>             # wait for the session space info to be updated to meta service
>             session.execute(
>                 'CREATE SPACE IF NOT EXISTS test_session(vid_type=FIXED_STRING(8)); USE test_session;'
>             )
>             time.sleep(10)
>             for i in range(0, 5):
>                 if i == 3:
>                     os.system('docker stop tests_graphd0_1')
>                     os.system('docker stop tests_graphd1_1')
>                     time.sleep(3)
>                 resp = session.execute('SHOW SESSIONS')
>                 assert resp.is_succeeded(), resp.error_msg()
>                 assert resp.space_name() == 'test_session' #<---------------------
>                 time.sleep(2)
>             session.release()
>             new_session = self.pool.get_session('root', 'nebula')
>             new_session.execute('SHOW SPACES')
>         except Exception as e:
> >           assert False, e
> E           AssertionError: AssertionError("assert '' == 'test_session' #-----------
> E               - test_session") 
> E           assert False
> ```
> 
> If that's the case, let's remove this assert from UT.

there's a test for multi-thread, it uses 50 threads to create space, causing the space is not enough for storaged. just fix it.

**Review by HarrisChu — APPROVED** [review]

**Pull request merged**